core/signals/handlers.py

```diff
@@ -0,0 +1,62 @@
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from clubManager import settings
+
+from common.major import get_major_from_netid
+from core.models import UserProfile
+from payments.models import PurchasedProduct
+
+import aiohttp
+from asgiref.sync import sync_to_async
+
+
+async def add_member_role(discord_id: int):
+    async with aiohttp.request(
+        "POST",
+        "http://localhost:2468/give-member-role",
+        json={"member_id": discord_id},
+        headers={"Authorization": f"Bearer {settings.API_SECRET}"},
+    ) as resp:
+        return await resp.json()
+
+
+@receiver(post_save, sender=UserProfile)
+async def update_roles_profile_signal(sender, instance: UserProfile, created, **kwargs):
```

Comment:
Just so we're on the same page - this signal is for automatically adding the Discord member role to a user once they link their Discord to their Club Manager account, correct?

It might be helpful to include a brief comment explaining what each of these signals does but not going to block merge for that

Reply:
Correct, thats a good idea